chain/ethereum/proto/codec.proto

```diff
@@ -0,0 +1,423 @@
+syntax = "proto3";
+
+package dfuse.ethereum.codec.v1;
+
+import "google/protobuf/timestamp.proto";
+
+message Block {
+  int32 ver = 1;
+  bytes hash = 2;
+  uint64 number = 3;
+  uint64 size = 4;
+  BlockHeader header = 5;
+  repeated BlockHeader uncles = 6;
+
+  repeated TransactionTrace transaction_traces = 10;
+  repeated BalanceChange balance_changes = 11;
+  repeated CodeChange code_changes = 20;
+
+  // Wheter or not a filtering process was run on this block. The filtering process populated the
+  // `transaction_traces` according to the `filtering_include_filter_expr` and
+  // `filtering_exclude_filter_expr` CEL expressions. A transaction will be present in the `transaction_traces`
+  // array if it matched the `filtering_include_filter_expr` and did *NOT* match the `filtering_exclude_filter_expr`.
+  //
+  // Moreover, each matching call that brought the transaction to be in `transaction_traces`
+  // array will have a `filtering_matched` flag set on it to broadcast the fact that this call
+  // match the inclusion/exclusion list.
+  //
+  // This flag controls all `transaction_traces` elements on the Block structure and on
+  // substructures if present.
+  bool filtering_applied = 40;
+
+  // The CEL filter expression used to include transaction in `transaction_traces` array, works
+  // in combination with `filtering_exclude_filter_expr` value.
+  string filtering_include_filter_expr = 41;
+
+  // The CEL filter expression used to exclude transaction in `transaction_traces` array, works
+  // in combination with `filtering_include_filter_expr` value.
+  string filtering_exclude_filter_expr = 42;
+}
+
+// BlockWithRefs is a lightweight block, with traces and transactions
+// purged from the `block` within, and only.  It is used in transports
+// to pass block data around.
+message BlockWithRefs {
+  string id = 1;
+  Block block = 2;
+  TransactionRefs transaction_trace_refs = 3;
+  bool irreversible = 4;
```

Comment:
Internally, StreamingFast systems for Ethereum uses -200 as the irreversible block. We added some arguments that enable tweaking the irreversible block (it cannot be passed 200 since in the backend, block after the 200 delta are not tracked anymore).

Not sure neither, that will need a live discussion. NEAR will be different has the chain has the concept of final block.